src/masternodes/mn_checks.cpp

```diff
@@ -1616,6 +1629,35 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         return !res ? res : mnview.ICXCloseMakeOfferTx(*offer, CICXMakeOffer::STATUS_CLOSED);
     }
 
+    Res operator()(const CLoanSetCollateralTokenMessage& obj) const {
+        auto res = CheckICXTx();
```

Comment:
Why loan need to check ICX tx?

Reply:
This checks that there are two vouts, I did raise this with Mihailo as there could technically only be one if there's no change. There's a fork update to change CheckICXTx change this to only check that the OP_RETURN output has a value of zero Satoshis.

Reply:
Should be resolved here.

https://github.com/DeFiCh/ain/pull/581/commits/731a77fbc1f9afc9345ae290cde9b9cc66ee45fb

Reply:
Will be changed to CheckCustomTx when I merge `setloantoken` PR.